### DISCUSSION TITLE
Is MITHRIL.js fit fot mobile development? have beachmark?

### DISCUSSION THREAD

**Discussion by anlexN** [MithrilJS/mithril.js#2731] [Q&A]

**Comment by lhorie** [comment]
Depends on what kind of mobile development.

If you're comparing with something like Ionic, yes, I've used Mithril for mobile and it runs fine. There are independent benchmarks elsewhere (e.g. [Stefan Krause's benchmark](https://www.stefankrause.net/js-frameworks-benchmark7/table.html)), which show that most frameworks these days have performance profiles suitable for mobile development (sub 3.0 geometric mean results). There's also some benchmarks in this repo. In the framework comparison page there's a comparison between implementations of dbmon - a stress test originally created by the Ember team - for [mithril](http://cdn.rawgit.com/MithrilJS/mithril.js/master/examples/dbmonster/mithril/index.html), [react](http://cdn.rawgit.com/MithrilJS/mithril.js/master/examples/dbmonster/react/index.html) and some other frameworks). You can run them from your phone to see what FPS count looks like for each.

Now, if you mean you're building a project dealing with things like sensors, you'll probably want to use the native SDK.

**Comment by cmnstmntmn** [comment]
if you plan to create a PWA, you can benefit of service workers; have a
look at pwa.rocks and test / compare to your own needs

**Comment by anlexN** [comment]
so @lhorie , what mobile development platform do you recommand ?

**Comment by lhorie** [comment]
Again, it depends on what you're doing. If you're doing a responsive thing that targets web and ios/android, something like cordova + mithril is hard to beat. React Native leaves quite a bit to be desired in terms of code reusability and I actually started noticing seeing various issues fairly early (animations stuttering, debugger disconnecting randomly, etc). I also experimented with weex, but found it very quirky and it produced terrible web code. There are some other newer offerings that I haven't evaluated yet, but from a glance they didn't seem much more advanced than the ones I did look into.

If you don't care about web at all, you might be better served by the native SDKs themselves (best end product quality, best debuggability, best employability, IMHO) or possibly something like Flutter (google backed, but haven't used myself, so take w/ a grain of salt).

**Comment by filisko** [comment]
I use Mithril.js with Cordova to consume a REST API amongst other stuff and runs very nice.

**Comment by mpfau** [comment]
We also use mithril as part of our native apps and are very happy with it. We needed to implement a workaround for improving fast rendering of large lists but besides that, we use pure mithril for all other parts of the UI.
